core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,497 @@
+                    logger.log(30, f"\tFit {index} ({fit_time}s): Current score {score} is better than best score {best_info['score']}. Updating model.")
+                    logger.log(30, f"\tOld # Features: {len(best_info['features'])} / New # Features: {len(candidate_features)}.")
+                    prev_best_model = best_info['model']
+                    best_info = {'model': curr_model, 'features': candidate_features, 'score': score, 'index': index}
+                    if not self.keep_models and prev_best_model.name != self.original_model.name:
+                        prev_best_model.delete_from_disk()
+                    self._debug_info['index_trajectory'].append(True)
+                else:
+                    logger.log(30, f"\tFit {index} ({fit_time}s): Current score {score} is not better than best score {best_info['score']}. Retrying.")
+                    if not self.keep_models:
+                        curr_model.delete_from_disk()
+                    self._debug_info['index_trajectory'].append(False)
+
+                if max_fits is not None and index >= max_fits:
+                    logger.log(30, f"\tReached maximum number of fits {max_fits}. Ending...")
+                    break
+                if index - best_info['index'] >= stop_threshold:
+                    logger.log(30, f"\tScore has not improved for {stop_threshold} iterations. Ending...")
+                    break
+                if self.time_limit <= self.model_fit_time + prune_time:
+                    logger.log(30, f"\tInsufficient time to finish next pruning round. Ending...")
+                    break
+        except TimeLimitExceeded:
+            logger.log(30, f"\tTime limit exceeded while pruning features. Ending...")
+        except Exception as e:
+            logger.error(traceback.format_exc())
+            logger.error(f"\tERROR: Exception raised during feature pruning. Reason: {e}. Ending...")
+            self._debug_info['exceptions'].append(str(e))
+            if self.raise_exception:
+                raise e
+
+        if auto_threshold:
+            best_info['features'] = [feature for feature in best_info['features'] if self.noise_prefix not in feature]
+        if not self.keep_models and best_info['model'].name != self.original_model.name:
+            best_info['model'].delete_from_disk()
+        self._debug_info['total_prune_time'] = self.original_time_limit - self.time_limit
+        self._debug_info['total_prune_fit_time'] = self._fit_time_elapsed
+        self._debug_info['total_prune_fi_time'] = self._fi_time_elapsed
+        self._debug_info['kept_ratio'] = len(best_info['features']) / len(original_features)
+        logger.log(30, f"\tSuccessfully ended prune loop after {index} iterations. Best score: {best_info['score']}.")
+        logger.log(30, f"\tFeature Count: {len(original_features)} -> {len(best_info['features'])} ({round(self.original_time_limit - self.time_limit, 2)}s)")
```

Comment:
"Count:" -> "Count before/after pruning:"

Reply:
should `original_features` and reduced features be somehow both stored in the model's `get_info()` for full reproducibility/informativeness?